packages/jest-config/src/index.js

```diff
@@ -19,12 +19,17 @@ import normalize from './normalize';
 function readConfig(
   argv: Argv,
   packageRoot: string,
+  // Whether it needs to look into `--config` arg passed to CLI.
+  // It only used to read initial config. If the initial config contains
+  // `project` property, we don't want to read `--config` value and rather
+  // read individual configs for every project.
+  dontUseArgvConfig?: boolean,
```

Comment:
i can't really come up with a more descriptive name. renamed it to `skipArgvConfigOption`. I hope that works!